teleport_tube.lua

```diff
@@ -270,14 +270,32 @@ local function repair_tube(pos, node)
 	update_meta(meta)
 end
 
-local function digiline_action(pos, _, channel, msg)
+local function digiline_action(pos, _, digiline_channel, msg)
 	local meta = minetest.get_meta(pos)
-	if channel ~= meta:get_string("digiline_channel") or type(msg) ~= "string" then
+	if digiline_channel ~= meta:get_string("digiline_channel") then
 		return
 	end
-	local name = meta:get_string("owner")
-	local cr = meta:get_int("can_receive")
-	update_tube(pos, msg, cr, name)
+	local channel = meta:get_string("channel")
+	local can_receive = meta:get_int("can_receive")
+	local t_msg = type(msg)
+	if t_msg == "string" then
+		channel = msg
+	elseif t_msg == "table" then
+		if type(msg.channel) == "string" then
+			channel = msg.channel
+		end
+		if nil ~= msg.can_receive then
+			if msg.can_receive == 1 or msg.can_receive == true then
+				can_receive = 1
+			else
+				can_receive = 0
+			end
+		end
+	else
+		return
+	end
```

Comment:
```suggestion
	if type(msg) == "string" then
		channel = msg
	elseif type(msg) == "table" then
		if type(msg.channel) == "string" then
			channel = msg.channel
		end
		if msg.can_receive == 1 or msg.can_receive == true then
			can_receive = 1
		elseif msg.can_receive == 0 or msg.can_receive == false then
			can_receive = 0
		end
	else
		return
	end
```

Reply:
why prefer calling type() multiple times from assigning a var?  
is the reason for readability or for performance?

Reply:
Probably no real difference with performance but IMO calling type grants better readability and is easier to follow.

For performance: introducing new local variable also has little performance impact, it is not instant operation. Static strings are very well optimized and type should also be very cheap operation.
I wouldn't be surprised if local variable would actually be slower but it does not really matter, in actual game difference probably cannot be even measured.

Reply:
Yeah, `type()` is such a small function that calling it repeatedly is better for readability than using a variable. And like @S-S-X said, there's practically no performance difference.